### PR TITLE
fix: Use correct logic to detect macOS target

### PR DESCRIPTION
**Pull request by Swatinem**

### CODE REVIEW

File: tests/cmake.py [hunk 1]
```diff
@@ -210,13 +210,7 @@ def cmake(cwd, targets, options=None):
...
         ]
         print("{} > {}".format(cwd, " ".join(checkcmd)), flush=True)
-        child = subprocess.run(checkcmd, stdout=subprocess.PIPE, cwd=cwd, check=True)
-        sys.stdout.buffer.write(child.stdout)
-        marker = b"Total number of reports: "
-        errors = child.stdout[child.stdout.rfind(marker) + len(marker) :]
-        errors = int(errors[: errors.find(b"\n")])
-        if errors > 0:
-            pytest.fail("code-checker analysis failed")
+        child = subprocess.run(checkcmd, cwd=cwd, check=True)
```
Comment (flub) [inline tests/cmake.py:213]: why is this changing?

**Review by flub — APPROVED** [review]

**Comment by lmdsp** [comment]
Thanks for fixing this. 
May I suggest using 

```cpp
if (__builtin_available(macOS 10.14, *)) 
{
    return backtrace_from_fp(addr, ptrs, max_frames);
}
```

This allows the best of both worlds, as it will detect the presence of the function at runtime,
depending on the host OS version instead of the minimum macOS deployment target.

**Comment by Swatinem** [comment]
@lmdsp thanks for the suggestion. I added this and also an explicit test for unwinding from a frame pointer that also triggers this.

File: tests/cmake.py (hunk 1 above)
Comment (Swatinem) [reply tests/cmake.py:213]: ah, code-checker now actually has a return code, so we don’t need to parse its stdout anymore ;-)

**Review by flub — APPROVED** [review]

**Pull request merged**